File: 946-validate-stack-sequences/946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) 
    {
        int n=pushed.size();
        int i=0;
        int j=0;
        for(auto &it:pushed)
        {
            pushed[i++]=it;
            while(i&&pushed[i-1]==popped[j])
            {
                i--;
                j++;
            }
        }
        return i==0;
    }
};

File: 946-validate-stack-sequences/946-validate-stack-sequences.js
/**
 * @param {number[]} pushed
 * @param {number[]} popped
 * @return {boolean}
 */
var validateStackSequences = function(pushed, popped) {
    let n=pushed.length;
    let i=0;
    let j=0;
    for(let it of pushed){
        pushed[i++]=it;
        while(i>0&&pushed[i-1]==popped[j]){
                i--;
                j++;
            }
    }
    return i==0;
};